compressor/base.py

```diff
@@ -32,21 +31,36 @@ class Compressor(object):
     depending implementations details.
     """
 
-    def __init__(self, content=None, output_prefix=None,
+    output_prefix = 'compressed'
+    output_mimetypes = {}
+
+    def __init__(self, resource_kind, content=None, output_prefix=None,
```

Comment:
what's the difference between resource_kind and output_prefix in practice? specifically, I am wondering whether there is any good reason to create e.g. a JsCompressor with a first arguent different from 'js'?

Reply:
...i now understand better that output_prefix is meant to control the output directory, which is now `compressed` by default for all filetypes other than js and css. how about making resource_kind another class variable?

Reply:
Good catch on `output_prefix` generally being the same as `resource_kind`,
I hadn't noticed about that.

I opted for making `resource_kind` an argument since it's
value is essentially controlled by the user in their settings.py.
Hardcoding it here would couple the kind to the compressor type,
which would probably result in breaking things.

It would also allow for running multiple version of one compressor
with different configurations, or using eg. a XmlCompressor
to compress generic xml, as well as html and svg.
Although I'm not sure how useful that would be in practice.

Since `output_prefix` is only used to vary the output path,
we could make it default to `resource_kind`
and remove the default class attribute.

Reply:
makes sense and agreed.